src/panel/poi/PoiPanel.jsx

```diff
@@ -44,6 +44,7 @@ export default class PoiPanel extends React.Component {
 
   componentDidMount() {
     fire('mobile_direction_button_visibility', false);
+    this.geolocateControlClickHandler = listen('geolocateControlClick', this.openDirection);
```

Comment:
What about 
```suggestion
    fire('set_current_poi', poi)
```
to notify other elements/buttons about the current poi ?

Note that this event would also be necessary in `componentDidUpdate`